app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+
+	def new; end
+	
+	def create	
+		auth_hash = request.env['omniauth.auth']
+		if auth_hash["uid"]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+        redirect_to root_path
+      else
+        redirect_to root_path, notice: "Failed to save the user"
+      end
+    else
+     	redirect_to root_path, notice: "Failed to authenticate"
+     end
+	end
+
+	def destroy
+		session.delete :user_id
+    redirect_to root_path
+	end
+end
```

Comment:
Your indenting seems a little funky in this file. Indenting code correctly makes it easier to read and follow. Indentation in all your other files looks good, though.